packages/apollo-gateway/src/index.ts

```diff
@@ -239,37 +245,50 @@ export class ApolloGateway implements GraphQLService {
     };
   }
 
-  protected async updateComposition(options?: {
-    engine?: GraphQLServiceEngineConfig;
-  }) {
+  private async updateComposition(): Promise<void> {
     const previousSchema = this.schema;
     const previousServiceDefinitions = this.serviceDefinitions;
     const previousCompositionMetadata = this.compositionMetadata;
 
-    if (options && options.engine) {
-      if (!options.engine.graphVariant)
-        console.warn('No graph variant provided. Defaulting to `current`.');
-      this.engineConfig = options.engine;
+    let result: Await<ReturnType<Experimental_UpdateServiceDefinitions>>;
+    try {
+      this.logger.debug('Loading configuration for gateway');
+      result = await this.updateServiceDefinitions(this.config);
+    } catch (e) {
+      this.logger.warn(
+        'Error checking for schema updates. Falling back to existing schema.',
+        e,
+      );
+      return;
     }
 
-    this.logger.debug('Loading configuration for gateway');
-    const result = await this.updateServiceDefinitions(this.config);
-
-    this.logger.debug('Configuration loaded for gateway');
-
-    if (!result.isNewSchema) return;
+    if (!('serviceDefinitions' in result)) return;
 
     if (
       JSON.stringify(this.serviceDefinitions) ===
```

Comment:
Hmm, the above conditional which checks for the presence of `serviceDefinitions` in `result` seems like it could be merged with this one?  Is that not the case?

i.e. delete the above `if` and:

```suggestion
      !('serviceDefinitions' in result) ||
      JSON.stringify(this.serviceDefinitions) ===
```

Reply:
Agree! Appreciate the commit suggestion but I went ahead and did it before I realized it was here 😄

Reply:
Resolved in 3cdc4b9